src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -182,6 +186,95 @@ public static List<string> GetHeaderValues(HttpRequestHeaders headers, string he
             return list;
         }
 
+        private static bool IsWithinRetryRange(HttpStatusCode statusCode)
+        {
+            // Retry on http 'server error' codes (5xx) as well as redirect
+
+            var rawStatus = (int)statusCode;
+            if (rawStatus == 302)
+                return true;
+            else if (rawStatus >= 500 && rawStatus <= 599)
```

Comment:
4xx errors are _usually_ less transient. But I do feel like we've seen on at least one occasion a 404 that wasn't accurate.  4xx error _definitely_ aren't success, so it shouldn't... hurt anything.